180 eagerness off

tszmytka opened the pull request:
This fixes #180

My initial idea of storing the meant-to-be-processed log line numbers would become more complicated than I initially anticipated so I changed it in favor of indicating only the log lines where the `ansicolor-plugin` should either begin or finish processing.

Main bullet points:
- Every time an `ansiColor` step is invoked a `ColorizedAction` step is added to the run context (to indicate starting and stopping)
- At the same time an html-invisible (`div`) marker is injected into the log output to allow `ColorConsoleAnnotator` to recognize when it should do something
- `ColorConsoleAnnotator` inspects each line and if it encounters the marker (matched with a corresponding action) - it either starts/stops its main functionality or simply continues assuming previously configured settings


All tests are passing (including my manual ones) but as there are not that many of them, I would appreciate another opinion.

dblock commented:
This looks great! I can't find anything wrong with the code ;) Just needs CHANGELOG/README updates and I say go for it. Would bump major version.

tszmytka commented:
> Just needs CHANGELOG/README updates and I say go for it. Would bump major version.

I keep forgetting about that. Now it is also there. 
I've also added a README example for the feature and moved the build wrapper pipeline usage before the old-fashioned UI defined job as this probably the preferred way of using it in `Jenkinsfile` times.
